### PR TITLE
Fix double-escaping because strip_html not marking string as Markup.

### LINKED ISSUE #5230: Broken character escaping on frontpage
untitled-1

`<p class="desc vtruncate" data-oldtext="Point &amp;amp;amp; click to forbid&amp;#x2F;allow any class of requests made by your browser. Use it to block scripts, iframes, ads, facebook, etc.">Point &amp;amp; click to forbid/allow any class of requests made by your brows…</p>`

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Fixes #5230

### CODE REVIEW

**Comment by eviljeff** [comment]
IMO we should just drop `|strip_html` entirely and use `|bleach` or similar.

**Comment by EnTeQuAk** [comment]
@eviljeff did that, tests should look good now. Mind another r?

File: src/olympia/legacy_api/templates/legacy_api/includes/addon.xml [hunk 1]
```diff
@@ -26,8 +26,8 @@
     {%- endfor %}
     {% endif %}
   </authors>
-  <summary>{{ addon.summary|strip_html(new_api) }}</summary>
-  <description>{{ addon.description|strip_html(new_api) }}</description>
+  <summary>{%- if new_api -%}{{ addon.summary }}{% else %}{{ addon.summary|clean(true) }}{% endif -%}</summary>
```
Comment (EnTeQuAk) [inline src/olympia/legacy_api/templates/legacy_api/includes/addon.xml]: you and me both 😆

**Comment by EnTeQuAk** [comment]
>  tbh I'm of two minds about overloading |clean - the strip_all_html parameter makes the filter do different things so should it be a separate filter? But bikeshedding so r+

My argument was that it's either more passive, just escaping things or, if the argument is given, the filter behaves more aggressive and removes what it'd escape. We can improve this over time though I think, it's way less hacky imho so I'll merge this but feel free to revert if any problems arise.

File: src/olympia/legacy_api/templates/legacy_api/includes/addon.xml (hunk 1 above)
Comment (diox) [inline src/olympia/legacy_api/templates/legacy_api/includes/addon.xml]: this is pretty old, see above: `{% set new_api = (api_version >= 1.5) %}`. Not sure if any clients are using a version older than 1.5 (which is the current version for the legacy API)...

**Pull request merged**